src/com/sony/simdetect/BootCompletedReceiver.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2019-2020 The LineageOS Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Rewritten in Kotlin by Pavel Dubrova <pashadubrova@gmail.com>
+ */
+
+package com.sony.simdetect
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+
+class BootCompletedReceiver : BroadcastReceiver() {
+    private var TAG = "SimDetect"
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if (context == null || intent == null) {
```

Comment:
@bartcubbins Can you make `context` non-nullable and get rid of the null-checks? Then this PR is ready to merge IMO.

Reply:
sure, will update pr asap